codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -275,16 +275,17 @@ private void generateOperationSerializerMiddleware(GenerationContext context, Op
             });
 
             writer.write("");
-
-            boolean withRestBindings = isOperationWithRestBindings(model, operation);
+            writer.write("request.Request.URL.Path = $S", httpTrait.getUri());
+            writer.write("request.Request.Method = $S", httpTrait.getMethod());
+            writer.write("restEncoder := rest.NewEncoder(request.Request)");
+            writer.write("");
 
             // we only generate an operations http bindings function if there are bindings
+            boolean withRestBindings = isOperationWithRestBindings(model, operation);
```

Comment:
Since you're not using this anywhere else anymore you should just embed it in the if